src/core/control/settings/Settings.cpp

```diff
@@ -44,7 +45,14 @@ constexpr auto DEFAULT_FONT_SIZE = 12;
     com = xmlNewComment((const xmlChar*)(var)); \
     xmlAddPrevSibling(xmlNode, com);
 
-Settings::Settings(fs::path filepath): filepath(std::move(filepath)) { loadDefault(); }
+Settings::Settings(SearchPath* configSearchPath): configSearchPath{configSearchPath} {
+    auto file = configSearchPath->findFile(SETTINGS_XML_FILE);
+    if (file.empty())
+        file = Util::getConfigFile(SETTINGS_XML_FILE);
+
+    this->filepath = file;
+    loadDefault();
+}
```

Comment:
I don't see any reason to search for `settings.xml` in the config path. The expectation is that the file will always be located in `$XDG_CONFIG_HOME/xournalpp/settings.xml` (or equivalent on non-Linux platforms). I also think this moves too much complexity into `Settings`, when finding the config file can be handled by the caller instead.